File: src/test/java/nl/wernerkroneman/Drawy/ModelEditor/EditorCommandTest.java
package nl.wernerkroneman.Drawy.ModelEditor;

import nl.wernerkroneman.Drawy.ModelEditor.CreateEntityEditorCommand;
import nl.wernerkroneman.Drawy.Modelling.CompositeModel;
import org.junit.Test;

public class EditorCommandTest {



    @Test(expected = RuntimeException.class)
    public void creationDouble() {

        CompositeModel model = new CompositeModel("Scene");

        CreateEntityEditorCommand stmt = new CreateEntityEditorCommand(() -> model);

        stmt.apply();
        stmt.apply();

    }

}

File: src/test/java/nl/wernerkroneman/Drawy/ModelEditor/EditorCommandTest.kt
/*
 * Copyright (c) 2017 Werner Kroneman
 *
 * This file is part of DrawyMcDrawface.
 *
 * DrawyMcDrawface is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * DrawyMcDrawface is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with DrawyMcDrawface.  If not, see <http://www.gnu.org/licenses/>.
 */

package nl.wernerkroneman.Drawy.ModelEditor

import nl.wernerkroneman.Drawy.Modelling.CompositeModel
import org.junit.Test

class EditorCommandTest {


    @Test(expected = RuntimeException::class)
    fun creationDouble() {

        val model = CompositeModel("Scene")

        val stmt = CreateEntityEditorCommand(target = { model })

        stmt.apply()
        stmt.apply()

    }

}